### PR TITLE
Add break-if validation tests

### LINKED ISSUE #1672: wgsl: validate `break-if` statement
* [ ] `fn (a) { break if true; }` invalid
* [ ] `loop { continuing { break if true; } }`
* [ ] `loop { continuing { break if (true); } }`
* [ ] `loop { var a = 1; continuing { break if true; a +=; } }` invalid, break if must be last
* [ ] `loop { break if true; }` invalid must be in continuing
* [ ] `switch 1 { case 1 { break if true; } }` invalid
* [ ] `for (;;) { break if true; }` invalid
* [ ] `while true { break if false; }` invalid
* [ ] `loop { continuing { break if 1 ^ 2; } }` invalid must be bool param
* [ ] `loop { continuing { break if vec(true, false); } }` invalid
* [ ] `loop { continuing { break if; } }` invalid missing condition
* [ ] `loop { continuing { break if (true; } }` invalid
* [ ] `loop { continuing { break if true); } }` invalid

### PR DESCRIPTION
**Pull request by dj2**
This CL adds a break-if validation set of tests. The one break-if tests in the `break` test file is moved over to the `break-if` test file.

Fixes #1672

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Review by zoddicus — COMMENTED** [review]
This looks reasonable from a CTS perspective, but I don't think I know the spec nuances well enought to sign off. Will leave for Ben to approve.

File: src/webgpu/shader/validation/parse/break_if.spec.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  loop_break: {
+    src: 'loop { break if true; }',
+    pass: false,
+  },
+  loop_if_break: {
+    src: 'loop { if true { break if false; } }',
+    pass: false,
+  },
+  continuing_break_if: {
+    src: 'loop { continuing { break if (true); } }',
```
Comment (ben-clayton) [inline src/webgpu/shader/validation/parse/break_if.spec.ts:19]: Drop the parentheses?

File: src/webgpu/shader/validation/parse/break_if.spec.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    pass: true,
+  },
+  continuing_break_if_not_last: {
+    src: 'loop { continuing { break if (true); let a = 4;} }',
+    pass: false,
+  },
+  while_break: {
+    src: 'while true { break if true; }',
```
Comment (ben-clayton) [inline src/webgpu/shader/validation/parse/break_if.spec.ts:27]: Suggest adding some variants in here where you keep the passing form (_[loop]_ → `continuing` → `break-if`), but swap the `loop` with `for` and `while`. e.g:

```
while true { continuing { break if true; } }
for (;;) { continuing { break if true; } }
```

**Review by ben-clayton — APPROVED** [review]
LGTM. + couple of suggestions.

File: src/webgpu/shader/validation/parse/break_if.spec.ts (hunk 1 above)
Comment (dj2) [reply src/webgpu/shader/validation/parse/break_if.spec.ts:23]: You can drop the `()`s, but it's also valid for `break if` to use them. So, I think we should have tests which have both styles.

I added a second test which does the same thing without parens.

Comment (ben-clayton) [reply src/webgpu/shader/validation/parse/break_if.spec.ts:23]: LGTM, thanks

File: src/webgpu/shader/validation/parse/break_if.spec.ts (hunk 2 above)
Comment (ben-clayton) [reply src/webgpu/shader/validation/parse/break_if.spec.ts:31]: I don't feel strongly here.

**Review by ben-clayton — APPROVED** [review]
LGTM!

**Pull request merged**